apps/rush-lib/src/logic/buildCache/ProjectBuildCache.ts

```diff
@@ -194,61 +229,103 @@ export class ProjectBuildCache {
     }
 
     terminal.writeVerboseLine(`Caching build output folders: ${filteredOutputFolders.join(', ')}`);
-    let encounteredTarErrors: boolean = false;
-    const tarStream: stream.Readable = tar.create(
-      {
-        gzip: true,
-        portable: true,
-        strict: true,
-        cwd: projectFolderPath,
-        filter: (tarPath: string, stat: tar.FileStat) => {
-          const tempStats: fs.Stats = new fs.Stats();
-          tempStats.mode = stat.mode;
-          if (tempStats.isSymbolicLink()) {
-            terminal.writeError(`Unable to include "${tarPath}" in build cache. It is a symbolic link.`);
-            encounteredTarErrors = true;
-            return false;
-          } else {
-            return true;
-          }
+    const tarExecutablePath: string | undefined = ProjectBuildCache._tarExecutablePath;
+    let localCacheEntryPath: string | undefined;
+
+    if (tarExecutablePath) {
+      const tempLocalCacheEntryPath: string = this._localBuildCacheProvider.getCacheEntryPath(cacheId);
+      const childProcess: ChildProcess = Executable.spawn(
+        tarExecutablePath,
+        ['-c', '-f', tempLocalCacheEntryPath, '-z', ...filteredOutputFolders],
+        {
+          currentWorkingDirectory: projectFolderPath
         }
-      },
-      filteredOutputFolders
-    );
-    const cacheEntryBuffer: Buffer = await this._readStreamToBufferAsync(tarStream);
-    if (encounteredTarErrors) {
-      return false;
+      );
+      const writeLocalCacheSuccess: boolean = await new Promise((resolve: (result: boolean) => void) => {
+        childProcess.on('exit', (code) => {
+          resolve(code === 0);
+        });
+      });
+
+      if (writeLocalCacheSuccess) {
+        localCacheEntryPath = tempLocalCacheEntryPath;
+      }
     }
 
-    const setLocalCacheEntryPromise: Promise<boolean> = this._localBuildCacheProvider.trySetCacheEntryBufferAsync(
-      terminal,
-      cacheId,
-      cacheEntryBuffer
-    );
+    let cacheEntryBuffer: Buffer | undefined;
+    let setLocalCacheEntryPromise: Promise<string> | undefined;
+    if (!localCacheEntryPath) {
+      // If we weren't able to create the cache entry with tar, try to do it with the "tar" NPM package
+      let encounteredTarErrors: boolean = false;
+      const tarStream: stream.Readable = tar.create(
+        {
+          gzip: true,
+          portable: true,
+          strict: true,
+          cwd: projectFolderPath,
+          filter: (tarPath: string, stat: tar.FileStat) => {
+            const tempStats: fs.Stats = new fs.Stats();
```

Comment:
What happens if an exception is thrown inside this callback?

Reply:
Rush crashes, even if this code is wrapped in a try/catch block. The `tar` package is kinda funky and doesn't handle issues very gracefully.

Reply:
Couldn't you do something like:
```ts
          cwd: projectFolderPath,
          filter: (tarPath: string, stat: tar.FileStat) => {
            try {
              const tempStats: fs.Stats = new fs.Stats();
              tempStats.mode = stat.mode;
              if (tempStats.isSymbolicLink()) {
                terminal.writeError(`Unable to include "${tarPath}" in build cache. It is a symbolic link.`);
                encounteredTarErrors = true;
                return false;
              } else {
                return true;
              }
            } catch (error) {
              terminal.writeError(`Error filtering tar paths: ` + error.toString());
              encounteredTarErrors = true;
              return false;
            }
          }
```